code/modules/art/paintings.dm

```diff
@@ -187,10 +255,25 @@
 	else if(istype(I, /obj/item/soap) || istype(I, /obj/item/reagent_containers/glass/rag))
 		return canvas_color
 
+/// Generates medium description
+/obj/item/canvas/proc/get_paint_tool_medium(obj/item/I)
+	if(!I)
+		return
+	if(istype(I, /obj/item/paint_palette))
+		return "Oil on canvas"
+	if(istype(I, /obj/item/toy/crayon))
+		return "Crayon on canvas"
+	else if(istype(I, /obj/item/pen))
+		return "Ink on canvas"
```

Comment:
why do you do if twice and then an else if

also its missing spraycans (should be before crayons because its a subtype of them)